elisp/haskell-ide-engine.el

```diff
@@ -145,13 +151,38 @@ association lists and count on HIE to use default values there."
    '()))
 
 (defun hie-handle-message (json)
-  (message "%s" (cdr (assq 'contents json))))
+  (message (format "%s" json)))
+
+(defun hie-handle-command-detail (json)
+  (let ((context
+         (hie-get-context (cdr (assq 'contexts json)))))
+    (setq haskell-ide-engine-process-handle-message
+          #'hie-handle-message)
+    (haskell-ide-engine-post-message
+     `(("cmd" . ,(hie-format-cmd haskell-ide-engine-current-cmd))
+       ("params" . ,context)))))
+
+(defun hie-format-cmd (cmd)
+  (format "%s:%s" (car cmd) (cdr cmd)))
+
+(defun hie-get-context (context)
+  (let ((start (save-excursion (goto-char (region-beginning))
+                               (list (line-number-at-pos) (current-column))))
+        (end (save-excursion (goto-char (region-end))
+                             (list (line-number-at-pos) (current-column))))
+        (filename (buffer-file-name)))
+    `(("file" . (("file" . ,filename)))
+      ("start_pos" . (("pos" . ,(apply 'vector start))))
+      ("end_pos" . (("pos" . ,(apply 'vector end)))))))
 
 (defun hie-run-command (plugin command)
   (setq haskell-ide-engine-process-handle-message
```

Comment:
Definitely, but we can do that separately imho. Moving in small steps is easier and faster.